### PR TITLE
GEOMESA-990 Not creating catalog table until needed

### PR DESCRIPTION
**Pull request by elahrvivaz**
Signed-off-by: Emilio Lahr-Vivaz <elahrvivaz@ccri.com>

### CODE REVIEW

File: geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/GeoMesaMetadata.scala [hunk 1]
```diff
@@ -103,16 +116,15 @@ class AccumuloBackedMetadata(connector: Connector,
...
-  }
+  override def delete(featureName: String, numThreads: Int): Unit =
+    if (tableLock.synchronized(tableExists)) {
+      val range = new Range(getMetadataRowKey(featureName))
+      val auths = authorizationsProvider.getAuthorizations
+      val deleter = connector.createBatchDeleter(catalogTable, auths, numThreads, metadataBWConfig)
+      deleter.setRanges(List(range))
+      deleter.delete()
+      deleter.close()
+    }
```
Comment (jahhulbert-ccri) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/GeoMesaMetadata.scala]: should we log a warn or info if the table does not exist?

Comment (elahrvivaz) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/GeoMesaMetadata.scala]: yeah, i'll add that

Comment (elahrvivaz) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/GeoMesaMetadata.scala]: added.

**Comment by jahhulbert-ccri** [comment]
gonna test an ingest and merge this in unless someone else has comments

**Comment by jahhulbert-ccri** [comment]
b8e4f4e8758622d3aaba2671f8557976a4c85396

**Pull request closed without merging**